tests/test_postgresql.py

```diff
@@ -1258,6 +1258,30 @@ def test_inline_exclude_constraint_text(self):
             "name='TExclID'))",
         )
 
+    def test_drop_exclude_constraint(self):
+        """test for #1300"""
+
+        autogen_context = self.autogen_context
+
+        m = MetaData()
+        t = Table(
+            "TTable", m, Column("XColumn", String), Column("YColumn", String)
+        )
+
+        op_obj = ops.DropConstraintOp.from_constraint(
+            ExcludeConstraint(
+                (t.c.XColumn, ">"),
+                where=t.c.XColumn != 2,
+                using="gist",
+                name="t_excl_x",
+            )
+        )
+
+        eq_ignore_whitespace(
```

Comment:
So we dont have a test that actually runs the op.drop_constraint() with generic and it turns out this fails, because we have another lookup in https://github.com/sqlalchemy/alembic/blob/dbdec2661b8a01132ea3f7a027f85fed2eaf5e54/alembic/operations/schemaobj.py#L178 that only expects `None` as the outlier type.

So let's add this test:

```diff
index a91d7f9..f0db669 100644
--- a/tests/test_postgresql.py
+++ b/tests/test_postgresql.py
@@ -169,6 +169,11 @@ class PostgresqlOpTest(TestBase):
             "WHERE (x > 5)"
         )
 
+    def test_drop_exclude_or_other_constraint(self):
+        context = op_fixture("postgresql")
+        op.drop_constraint("t_excl_x", "TTable", type_=None)
+        context.assert_('ALTER TABLE "TTable" DROP CONSTRAINT t_excl_x')
+
     def test_create_exclude_constraint_quoted_literal(self):
         context = op_fixture("postgresql")
         op.create_exclude_constraint(
```

Reply:
I just found exactly the same thing and added almost exactly the same test in my latest commit.